libsolidity/ast/Types.cpp

```diff
@@ -362,6 +366,59 @@ MemberList::MemberMap Type::boundFunctions(Type const& _type, ASTNode const& _sc
 	if (auto refType = dynamic_cast<ReferenceType const*>(&_type))
 		typeLocation = refType->location();
 
+	return usingForDirectives | ranges::views::filter([&](UsingForDirective const* _directive) -> bool {
+		// Convert both types to pointers for comparison to see if the `using for`
+		// directive applies.
+		// Further down, we check more detailed for each function if `_type` is
+		// convertible to the function parameter type.
+		return
+			!_directive->typeName() ||
+			*TypeProvider::withLocationIfReference(typeLocation, &_type, true) ==
+			*TypeProvider::withLocationIfReference(
+				typeLocation,
+				_directive->typeName()->annotation().type,
+				true
+			);
+	}) | ranges::to<vector<UsingForDirective const*>>;
+}
+
+}
+
+FunctionDefinitionResult Type::userDefinedOperator(Token _token, ASTNode const& _scope, bool _unaryOperation) const
+{
+	if (!typeDefinition())
+		return nullptr;
+
+	set<FunctionDefinition const*> seenFunctions;
+	for (UsingForDirective const* ufd: usingForDirectivesForType(*this, _scope))
+		for (auto const& [pathPointer, operator_]: ufd->functionsAndOperators())
+		{
+			if (operator_ != _token)
+				continue;
+			FunctionDefinition const& function = dynamic_cast<FunctionDefinition const&>(
+				*pathPointer->annotation().referencedDeclaration
+			);
+			FunctionType const* functionType = dynamic_cast<FunctionType const*>(
+				function.libraryFunction() ? function.typeViaContractName() : function.type()
+			);
+			solAssert(functionType && !functionType->parameterTypes().empty());
+			solAssert(isImplicitlyConvertibleTo(*functionType->parameterTypes().front()));
+			if ((_unaryOperation && function.parameterList().parameters().size() == 1) ||
+				(!_unaryOperation && function.parameterList().parameters().size() == 2))
+				seenFunctions.insert(&function);
+		}
+
+	if (seenFunctions.size() == 1)
+		return *seenFunctions.begin();
+	else if (seenFunctions.size() == 0)
+		return FunctionDefinitionResult::err("A user-defined operator not found.");
```

Comment:
"Operator has not been user-defined." - but only display this error message if it is possible to have a user-defined operator for this type.

Reply:
Code updated. Now, missing operator error message is extended with an extra information about user-defined operator   only if this type is user defined and operator can be user implemented.